### PR TITLE
feat(devtools): patchNamedConstant — injectable TS-AST SourcePatcher (#566)

### PR DESCRIPTION
**Pull request by laofahai**
## What

Lands `patchNamedConstant` in `@linchkit/devtools` — the concrete TS-AST implementation of core's typescript-free `SourcePatcher` seam (merged in #590). This is the first brick of the #566 "say→change an existing code-condition rule's threshold and graduate it to real source" chain.

`patchNamedConstant` rewrites a single top-level `export const NAME = <literal>` initializer in an existing source file by splicing the initializer's byte span out of the original source string — the whole file is never re-emitted, so indentation, the `: number` annotation, the trailing `;`, and any trailing comments survive untouched.

- Walks **only** `sourceFile.statements` for an exported `VariableStatement` whose name matches → nested consts, comments, strings, and substring names (`X` vs `X_2`) are unmatchable.
- Throws `NOT FOUND` (0 matches), `AMBIGUOUS` (>1), `NO INITIALIZER` (e.g. `export declare const X: number`).
- Idempotent: returns `changed: false` **only** when the constant already holds the target value.
- Compile-time `const _: SourcePatcher = patchNamedConstant` guard confirms it satisfies core's contract.

`typescript` is promoted from a dev to a runtime dependency since the patcher ships in the package output.

## Why

Core declares the `SourcePatcher` seam but stays typescript-free (Minimal Core). The heavy TS-compiler-API patcher lives here, in devtools, and is injected into `ProposalFileWriter` at the graduation call-site (a follow-up PR wires the injection at `proposal-graduate-api.ts`). This is the deterministic, named-constant graduation primitive for "说→改既有规则".

## Tests

11 new pinned tests + the assignability guard; full devtools suite green. `bun run check` + `bun run typecheck` clean.

> Note on the full-suite gate: the local `bun run test` batched runner currently hangs in an **unrelated** `packages/cli` batch due to a leftover dev server bound to :3001/:3002 + proxy contamination — not this diff. The devtools+core blast radius (4520 tests / 247 files) passes clean in 6.8s. CI clean-install is authoritative.

Related: #566, #590

## Release Notes

* **New Features**
  * Added capability to patch exported TypeScript named constants while preserving surrounding comments, type annotations, indentation, and file structure.

* **Tests**
  * Comprehensive test suite added covering successful patching, edge cases, and error handling scenarios.

* **Dependencies**
  * TypeScript migrated from development to runtime dependencies.

### CODE REVIEW

File: packages/devtools/src/patch-named-constant.ts [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+ * first declaration's combined modifier flags also folds in `declare`, which is
+ * fine here — we only care whether the declaration is exported.
+ */
+function isExported(statement: ts.VariableStatement): boolean {
+  const declaration = statement.declarationList.declarations[0];
+  if (declaration === undefined) {
+    return false;
+  }
+  return (ts.getCombinedModifierFlags(declaration) & ts.ModifierFlags.Export) !== 0;
+}
```
[the remark this answers is not in the document]
Comment (laofahai) [reply packages/devtools/src/patch-named-constant.ts:44]: Done in 0780159d — `isExported` now reads `statement.modifiers` directly (no `getCombinedModifierFlags`), removing the parent-pointer dependency.

File: packages/devtools/src/patch-named-constant.ts [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+export function patchNamedConstant(request: SourcePatchRequest): SourcePatchResult {
+  const { source, constantName, newValueLiteral } = request;
+
+  // `setParentNodes = true` so `getStart(sourceFile)` works on nested nodes.
+  const sourceFile = ts.createSourceFile(
+    "patch-input.ts",
+    source,
+    ts.ScriptTarget.Latest,
+    /* setParentNodes */ true,
+  );
```
[the remark this answers is not in the document]
Comment (laofahai) [reply packages/devtools/src/patch-named-constant.ts]: Done in 0780159d — the parse now runs with `setParentNodes: false`; `initializer.getStart(sourceFile)` is passed the source file explicitly, so parent pointers are not needed.

File: packages/devtools/src/patch-named-constant.ts [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+  for (const statement of sourceFile.statements) {
+    if (!ts.isVariableStatement(statement)) {
+      continue;
+    }
+    if (!isExported(statement)) {
+      continue;
+    }
+
+    for (const declaration of statement.declarationList.declarations) {
+      // Match on the AST identifier — never on raw text — so comments, string
```
[the remark this answers is not in the document]
Comment (laofahai) [reply packages/devtools/src/patch-named-constant.ts:91]: Fixed in 0780159d — added a `(statement.declarationList.flags & ts.NodeFlags.Const)` guard so `export let` / `export var` no longer match the `export const`-only contract, plus export-let/export-var regression tests locking the boundary.

File: packages/devtools/src/patch-named-constant.ts [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+ * Conforms to core's `SourcePatcher` contract. On success returns the patched
+ * source plus the replaced literal; returns `changed: false` ONLY for an
+ * idempotent no-op (found, but the value already equals the target). THROWS
+ * when the constant is not found, is ambiguous (more than one match — we refuse
+ * to guess), or has no initializer.
+ */
+export function patchNamedConstant(request: SourcePatchRequest): SourcePatchResult {
+  const { source, constantName, newValueLiteral } = request;
+
+  // `setParentNodes = true` so `getStart(sourceFile)` works on nested nodes.
```
[the remark this answers is not in the document]
Comment (laofahai) [reply packages/devtools/src/patch-named-constant.ts]: Addressed in 0780159d — the misleading comment is gone; the parse now uses `setParentNodes: false` with a comment explaining that `getStart(sourceFile)` takes the source file explicitly (no parent pointers needed).
